Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -628,6 +628,10 @@ public class HTTPClient {
     public struct Configuration {
         /// TLS configuration, defaults to `TLSConfiguration.forClient()`.
         public var tlsConfiguration: Optional<TLSConfiguration>
+        #if canImport(Network)
+            /// TLS configuration, defaults to `TLSConfiguration.forClient()`.
```

Comment:
This comment looks wrong.

Reply:
> I'm curious as to why we need a different data structure. It seems like we could shim this under the existing `TLSConfiguration` structure from NIOSSL, couldn't we?

We can shim most of it. `SecCertificate` can be created from the DER data of a `NIOSSLCertificate`. `SecIdentity` is the problem case. It requires a p12 or a private key already in the keychain and currently there is no way to generate this from the combination of `NIOSSLPrivateKey` and `NIOSSLCertificate` unless I missed something.

Reply:
Ah yeah, SecIdentity is rough.

Perhaps this is the moment we encourage AHC to create its own representation of a TLSConfiguration that can abstract over some of these better.

Reply:
Can we expose BoringSSL pkcs#12 generation code in NIOSSL? If we have this, we can probably generate a SecIdentity from a TLSConfiguration.

Reply:
We could, though I don't love doing it.

Reply:
If we want a common structure for both NIOSSL and NIOTransportServices TLS setup I can't see any other way around supplying a p12 generation.

Reply:
Yes, I think I agree. I'd like to know what some other folks prefer here: @artemredkin @ktoso?

Reply:
I think common structure would be preferable over implementing an additional abstraction over NIOSSL and NIOTransportServices. I wonder, can that structure live in NIOTS?

Reply:
@artemredkin I believe with p12 generation from a NIOSSL private key and certificate chain we can use `TLSConfiguration` to cover most situations.

Reply:
>  I wonder, can that structure live in NIOTS?

I don't really want it to at this early stage.